src/pages/cityWeather/CityWeather.jsx

```diff
@@ -1,17 +1,21 @@
 import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import { publicApiInstance } from '../../utils/api';
-import { WeatherCard, SearchInput, SearchBar, Tooltip } from '../../components';
+import { WeatherCard, SearchInput, SearchBar, Tooltip, Button } from '../../components';
 import endpoints from '../../config/endpoints';
-import { minSearchCharacters } from '../../config/constants';
+import { minSearchCharacters, SAVE_SEARCH, DELETE_SEARCH } from '../../config/constants';
 import { translations } from '../../utils/translations';
 import * as S from './CityWeather.styles';
 
 const CityWeather = () => {
   const [cityName, setCityName] = useState('');
-  const [selectedCity, setSelectedCity] = useState('');
+  const [selectedCity, setSelectedCity] = useState({});
   const [results, setResults] = useState([]);
   const [chooseCityForecast, setChooseCityForecast] = useState([]);
 
+  const dispatch = useDispatch();
+  const searchResults = useSelector(state => state.savedSearch);
```

Comment:
please think about naming for variables
it's not clear what's the difference between `results` and `searchResults`, and `cityName` actually is not a city name, but input value to search

Reply:
It was really hard to give a new names for variables, but hope I changed correctly.